mutornadomon/collectors/ioloop_util.py

```diff
@@ -22,7 +22,15 @@ def run_timed_callback(callback):
         def add_timed_handler(fd, handler, events):
             def timed_handler(*args, **kwargs):
                 start_time = time.time()
+
+                if self.monitor.stats_init == True:
+                    self.monitor.profiler.enable()
+
                 result = handler(*args, **kwargs)
+
+                if self.monitor.stats_init == True:
```

Comment:
Check this against a local instead of the global. Remember that handler could be your http endpoint getting hit or your yield for changing the state of stats_init